km63/Buchynska_Kateryna/laboratory1.sql

```diff
@@ -1,2 +1,148 @@
 -- LABORATORY WORK 1
 -- BY Buchynska_Kateryna
+/*---------------------------------------------------------------------------
+1.Створити схему даних з назвою – прізвище студента, у якій користувач може: 
+робити вибірки з таблиць та оновлювати дані
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+CREATE USER buchynska IDENTIFIED BY buchynska;
+DEFAULT TABLESPACE USER;
+
+
+GRANT SELECT ANY TABLE to buchynska;
+GRANT UPDATE ANY TABLE to buchynska;
+GRANT CREATE TABLE to buchynska;
+
+GRANT CONNECT to buchynska;
+QUOTA 100M to buchynska; 
+
+
+
+
+
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+2. Створити таблиці, у яких визначити поля та типи. Головні та зовнішні ключі 
+створювати окремо від таблиць використовуючи команди ALTER TABLE. 
+На вулиці стоїть будинок, що має 10 квартир.
+4 бали
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+
+
+CREATE CREATE TABLE House
+(
+    house_address VARCHAR2(15)
+);
+
+CREATE TABLE Room 
+(
+    room_number INT (10);
+);
+
+
+
+CREATE TABLE Room_in_house 
+(
+ room_number INT(10),
+ house_number INT(10),
+);
+
+ALTER TABLE House ADD CONSTRAINT PRIMARY KEY  house_pk  (house_address);
+ALTER TABLE Room ADD CONSTRAINT PRIMARY KEY  room_pk  (room_number);
+ALTER TABLE Room_in_house ADD CONSTRAINT FOREIGN KEY  room_in_house_fk  (room_number, house_number);
+
+
+
+
+
+
+
+
+
+
+
+
+
+  
+/* --------------------------------------------------------------------------- 
+3. Надати додаткові права користувачеві (створеному у пункті № 1) для створення таблиць, 
+внесення даних у таблиці та виконання вибірок використовуючи команду ALTER/GRANT. 
+Згенерувати базу даних використовуючи код з теки OracleScript та виконати запити: 
+
+---------------------------------------------------------------------------*/
+--Код відповідь:
+
+GRANT ALTER TABLE to buchynska ;
+GRANT SELECT TABLE to buchynska ;
+
+
+
+
+
+
+/*---------------------------------------------------------------------------
+3.a. 
+Який номер замовлення куди входить найдорожчий товар та яке ім'я покупця цього замовлення?
+Виконати завдання в алгебрі Кодда.
+4 бали
+---------------------------------------------------------------------------*/
+
+--Код відповідь:
+
+PROJECT(TIMES (order_price=max(order_price)) order_num, cust_name);
+
```

Comment:
SELECT CUSTOMERS.cust_name, ORDERS.order_num
FROM customers,orders,orderitems
WHERE item_price in(
SELECT max(item_price)
FROM orderitems
)
AND customers.cust_id = orders.cust_id
AND orders.order_num = orderitems.order_num;